src/BuildQueue.cpp

```diff
@@ -1248,30 +1250,16 @@ namespace t2
     return state;
   }
 
-  bool HasBuildStoppingFailures(const BuildQueue* queue)
-  {
-    if (queue->m_FailedNodeCount > 0)
-    {
-      if (0 == (queue->m_Config.m_Flags & BuildQueueConfig::kFlagContinueOnError))
-      {
-        return true;
-      }
-    }
-    return false;
-  }
-
   static bool ShouldKeepBuilding(BuildQueue* queue)
   {
-    // Stop running if we were signalled
-    if (nullptr != SignalGetReason())
+    // If we're quitting, definitely stop building.
+    if (queue->m_MainThreadWantsToCleanUp)
       return false;
 
-    // Stop running if there are errors and we're stopping on the first error.
-    if (HasBuildStoppingFailures(queue))
-      return false;
 
-    // If we're quitting, definitely stop building.
-    if (queue->m_QuitSignalled)
+    //you'd think we don't have to check for this, as the main thread will realize the build has failed, and will shut us down,
+    //but if we don't check this in the buildloop, we'll actually continue to build nodes whose dependencies have failed.
+    if (queue->m_FailedNodeCount > 0)
```

Comment:
Should this respect kFlagContinueOnError?

Reply:
kFlagContinueOnError has never worked.  either we broke it very early when we adopted tundra, or it never worked.   It would be great to have the feature again, but I didn't care to implement it in this pr.